nio-core/src/main/scala/zio/nio/core/channels/DatagramChannel.scala

```diff
@@ -1,42 +1,148 @@
 package zio.nio.core.channels
 
 import java.io.IOException
+import java.net.{ DatagramSocket => JDatagramSocket, SocketAddress => JSocketAddress }
 import java.nio.channels.{ DatagramChannel => JDatagramChannel }
-import java.net.{ DatagramSocket => JDatagramSocket }
 
-import zio.nio.core.{ ByteBuffer, SocketAddress }
 import zio.{ IO, UIO }
+import zio.nio.core.{ ByteBuffer, SocketAddress, SocketOption }
 
+/**
+ * A [[java.nio.channels.DatagramChannel]] wrapper allowing for basic [[zio.ZIO]] interoperability.
+ */
 final class DatagramChannel private[channels] (override protected[channels] val channel: JDatagramChannel)
     extends GatheringByteChannel
     with ScatteringByteChannel {
 
-  def bind(local: SocketAddress): IO[IOException, DatagramChannel] =
-    IO.effect(new DatagramChannel(channel.bind(local.jSocketAddress))).refineToOrDie[IOException]
+  private def bind(local: Option[SocketAddress]): IO[IOException, DatagramChannel] = {
+    val addr: JSocketAddress = local.map(_.jSocketAddress).getOrElse(null)
+    IO.effect(channel.bind(addr)).as(this).refineToOrDie[IOException]
+  }
 
-  def connect(remote: SocketAddress): IO[IOException, DatagramChannel] =
-    IO.effect(new DatagramChannel(channel.connect(remote.jSocketAddress))).refineToOrDie[IOException]
+  private def connect(remote: SocketAddress): IO[IOException, DatagramChannel] =
```

Comment:
in core we have un-opinionated so please do not make change to the inteface. we need to keep it close to original.

Reply:
Done.